apps/discordbot/bot.py

```diff
@@ -188,6 +197,63 @@ def promote_user_to_hoser(self, tag):
             return True
         return False
 
+    def event_announcement(self):
+        print('Announcements Thread started...')
+        
+        times_msg = {
+            "week": "NEXT WEEK",
+            "today": "TODAY",
+            "tomorrow": "TOMORROW",
+            "hour": "IN 1 HOUR",
+            "now": "NOW",
+        }
+        
+        def announcer(time_before):
+
+            events = event_checker(time_before)
+
+            if events:
+                msg = f"**What's happening {times_msg[time_before]}**"
+                asyncio.run_coroutine_threadsafe(
+                        self.client.get_channel(ANNOUNCEMENTS_CHANNEL_ID).send(msg), self.loop
+                    ).result(TIMEOUT_SECS)
+                print('hey hey hey time to check') # debugging
+
+                send_embed(events)
+        
+        def send_embed(events):
+            for event in events:
+                embed = discord.Embed(
+                    title=event.name,
+                    description=event.description,
+                    colour=discord.Colour.red()
+                )
+                embed.add_field(name='Date', value=event.date, inline=True)
+                embed.add_field(name='Time', value=event.get_time_string())
+                embed.add_field(name='Link', value=event.link)
+                asyncio.run_coroutine_threadsafe(
+                    self.client.get_channel(ANNOUNCEMENTS_CHANNEL_ID).send(embed=embed), self.loop
+                ).result(TIMEOUT_SECS)
+
+        
+        schedule.every().sunday.at("17:00").do(partial(announcer, "week"))
```

Comment:
I don't like the fact that we're using magic strings here. If "week" or "tomorrow" or "today" get accidentally misspelled, we might end up with a bug that we won't notice until the time comes around. My suggestion would be to define some constants like  TODAY = "today" and just use those instead.

Reply:
Added the constants